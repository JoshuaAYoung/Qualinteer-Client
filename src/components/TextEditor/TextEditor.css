.jodit_vable_theme .jodit_toolbar {
  background: #f4bd5b;
  border: none;
}
.jodit_vable_theme .jodit_statusbar {
  background: #f4bd5b;
  border: none;
}

.jodit_placeholder {
  font-family: 'Montserrat', sans-serif;
  color: #777777;
}

.jodit_toolbar_btn-separator {
  border-right: 1px solid #777777 !important;
}

.jodit_tooltip {
  display: none;
}

.jodit_tooltip.jodit_tooltip_visible {
  display: initial;
}
