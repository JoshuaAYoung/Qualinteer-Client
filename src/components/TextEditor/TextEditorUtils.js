export default {
  readonly: false,
  allowResizeY: true,
  editorCssClass: 'textEditor',
  height: 400,
  theme: 'vable',
  placeholder: "<span style='font-size: 18px;'><strong>Opportunity Description</strong></span><br><br><strong>Give potential volunteers more information about the position.</strong><br><br>What's expected? Who will they report to? Will they be working with other volunteers or is the work performed solo?<br><br>Format your text however you'd like. Add a heading. Insert a list of tasks. Add your organization's logo. Maybe add a link to your organization's about page. Anything that you think might be helpful to <u>catch the eye</u> of a potential volunteer!",
  buttons: [
    'source',
    'fullsize',
    '|',
    'bold',
    'strikethrough',
    'underline',
    'italic',
    'eraser',
    '|',
    'ul',
    'ol',
    '|',
    'outdent',
    'indent',
    '|',
    'font',
    'fontsize',
    'brush',
    'paragraph',
    '|',
    'image',
    'link',
    '|',
    'align',
    'undo',
    'redo',
    '|',
    'hr'
  ],
  //window size < 940 px
  buttonsMD: [
    'source',
    'fullsize',
    '|',
    'bold',
    'underline',
    'italic',
    '|',
    'ul',
    'ol',
    '|',
    'font',
    'fontsize',
    'brush',
    'paragraph',
    '|',
    'image',
    'link',
    '|',
    'align',
    'undo',
    'redo',
    '|',
    'hr',
    'dots'
  ],
  //window size < 740px
  buttonsSM: [
    'source',
    'fullsize',
    '|',
    'bold',
    'italic',
    '|',
    'ul',
    'ol',
    '|',
    'fontsize',
    'brush',
    'paragraph',
    '|',
    'image',
    'link',
    '|',
    'align',
    'undo',
    'redo',
    '|',
    'dots'
  ],
  //window size < 440px
  buttonsXS: [
    // 'source',
    'bold',
    'image',
    '|',
    'brush',
    'paragraph',
    'eraser',
    '|',
    'align',
    '|',
    'undo',
    'redo',
    '|',
    'dots'
  ]

}