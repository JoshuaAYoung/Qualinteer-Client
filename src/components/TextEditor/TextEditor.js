import React, { Component } from 'react';
import JoditEditor from "jodit-react";
import VableContext from '../../context/VableContext'
import TextEditorUtils from './TextEditorUtils'
import './TextEditor.css'
import ValidationError from '../../utilities/ValidationError'

export default class TextEditor extends Component {
  static contextType = VableContext;

  state = {
    touched: false,
  }

  constructor(props) {
    super(props)
    this.editor = null;
  }

  setContent = (description) => {
    this.context.setDescription({
      description
    })
    this.setState({
      touched: true
    })
  }

  generateError = () => {
    if (this.state.touched === true && !this.context.user.description) {
      return 'Please enter a description.';
    }
  }

  config = TextEditorUtils;

  render() {
    return (
      <div className='editorContainer'>
        <JoditEditor
          ref={editor => { this.editor = editor }}
          value={this.context.user.description}
          config={this.config}
          tabIndex={1}
          onBlur={description => this.setContent(description)}
        />
        {this.generateError() && <ValidationError message={this.generateError()} />}
      </div>
    );
  }
}