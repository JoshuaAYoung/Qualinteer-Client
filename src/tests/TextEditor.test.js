import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TextEditor from '../components/TextEditor/TextEditor';
import VableContext from '../context/VableContext';

describe('TextEditor Component', () => {
  it.skip('TextEditor Smoke Test', () => {
    const div = document.createElement('div');

    ReactDOM.render(<TextEditor />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('TextEditor Snapshot Test', () => {
    const wrapper = shallow(
      <VableContext.Provider
        value={{ user: { description: 'Test Description' } }}
      >
        <TextEditor />
      </VableContext.Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
