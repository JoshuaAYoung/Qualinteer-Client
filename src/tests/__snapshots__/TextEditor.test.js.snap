// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextEditor Component TextEditor Snapshot Test 1`] = `<TextEditor />`;
